build/archives/BUILD.gn

```diff
@@ -21,7 +21,28 @@ generated_file("artifacts_entitlement_config") {
   }
 }
 
-if (build_engine_artifacts && !is_android) {
+# This target is used to build a zip archive (artifacts.zip) of host tools such
+# as flutter_tester, the Dart kernel compiler, the impellerc shader compiler,
+# and other tooling that is bundled in debug-mode host builds.
+#
+# For historical reasons, this bundle also includes gen_snapshot (which is a
+# target-platform-specific binary) for desktop platforms, though the
+# un-suffixed gen_snapshot is unused on macOS (gen_snapshot_arm64 and
+# gen_snapshot_x64 are used in its place).
+#
+# Target-specific iOS artifacts.zip are produced in:
+# //flutter/sky/tools/create_full_ios_framework.py
+#
+# Target-specific Android archives are produced in:
+# //flutter/shell/platform/android:gen_snapshot
+#
+# macOS-specific gen_snapshot_arm64 and gen_snapshot_x64 are produced in:
+# //flutter/sky/tools/create_macos_gen_snapshots.py
+#
+# TODO: https://github.com/flutter/flutter/issues/38935
+# Target-specific binaries such as gen_snapshot should be separated into a
+# separate archive target from host binaries such as flutter_tester.
+if (build_engine_artifacts && !is_android && !is_ios) {
```

Comment:
I'm a little confused. Is this change in anticipation of `build_engine_artifacts` no longer excluding the iOS builds? Or for the host and iOS toolchains to be merged?

Reply:
This actually changes nothing and is just to make it more obvious to the reader that this target is excluded on both android and iOS. See the PR description for why it was already disabled on iOS.

Reply:
Updated the title and description a little for clarity. Really this PR is primarily about making it obvious we don't use this target on iOS builds. 

The doc comment is just so no one else has to go digging to figure out what this target is and isn't actually used for.

Reply:
Going to try moving the !is_android to the definition of this variable in common/config.gni and see what breaks.